kimchi/src/circuits/constraints.rs

```diff
@@ -380,17 +390,23 @@ impl<F: FftField + SquareRootField> LookupConstraintSystem<F> {
 }
 
 impl<F: FftField + SquareRootField> ConstraintSystem<F> {
-    /// creates a constraint system from a vector of gates ([CircuitGate]), some sponge parameters ([ArithmeticSpongeParams]), and the number of public inputs.
+    /// Creates a constraint system from a vector of gates ([CircuitGate]),
+    ///   some sponge parameters ([ArithmeticSpongeParams]),
+    ///   an optional foreign modulus, and the number of public inputs.
+    ///   Note that passing an empty vector for the foreign modulus parameter
+    ///   disables the foreign field multiplication gate.
     pub fn create(
         gates: Vec<CircuitGate<F>>,
         lookup_tables: Vec<LookupTable<F>>,
         fr_sponge_params: ArithmeticSpongeParams<F>,
+        foreign_modulus: Vec<F>,
```

Comment:
I think this should be an `Option<Vec<F>>`. Also it sounded like your field needs to be split in 3 `F` no? Would it make sense to use `[F; 3]` for this (and what's in prover/verifier index?)